usr/share/rear/rescue/default/900_clone_users_and_groups.sh

```diff
@@ -1,41 +1,60 @@
-# copy required system users and groups to the rescue system
 
-if [[ "$CLONE_ALL_USERS_GROUPS" =~ ^[yY1] ]]; then
-    CLONE_USERS=($(cut -d ':' -f '1' /etc/passwd))
-    CLONE_GROUPS=($(cut -d ':' -f '1' /etc/group))
+# Copy users and groups to the ReaR recovery system:
+
+if is_true "$CLONE_ALL_USERS_GROUPS" ; then
+    # In particular exclude a possible NIS extension line '+::::::'
+    CLONE_USERS=( $( grep -o '^[[:alnum:]_][^:]*' /etc/passwd ) )
+    # In particular exclude a possible NIS extension line '+:::'
+    CLONE_GROUPS=( $( grep -o '^[[:alnum:]_][^:]*' /etc/group ) )
 fi
 
-Log "Cloning users: ${CLONE_USERS[@]}"
-for u in "${CLONE_USERS[@]}" ; do
-	# go over all users
-	if pwd=$(getent passwd "$u") ; then
-		# pwd="daemon:x:2:2:Daemon:/sbin:/bin/bash"
-		# if the user exists, add to the passwd in rescue system
-		# skip if this user exists already in the rescue system
-		grep -q "^$pwd:" $ROOTFS_DIR/etc/passwd && continue
-		echo "$pwd" >>$ROOTFS_DIR/etc/passwd
-		# strip gid from passwd line
-		pwd="${pwd#*:*:*:}"
-		gid=${pwd%%:*}
-		# add gid to groups to collect
-		CLONE_GROUPS=( "${CLONE_GROUPS[@]}" "$gid" )
-	else
-		Debug "WARNING: Could not collect user info for '$u'"
-	fi
-done
+local user=""
+local passwd_entry=""
+local groupID=""
+local group_entry=""
+local group=""
 
-Log "Cloning groups: ${CLONE_GROUPS[@]}"
-for g in "${CLONE_GROUPS[@]}" ; do
-	# go over all users
-	if grp=$(getent group "$g") ; then
-		# grp="daemon:x:2:"
-		# if the group  exists, add to the group in rescue system
-		# skip if this user exists already in the rescue system
-		grep -q "^$grp" $ROOTFS_DIR/etc/group && continue
-		echo "$grp" >>$ROOTFS_DIR/etc/group
-	else
-		Debug "WARNING: Could not collect group info for '$g'"
-	fi
+# For the 'test' one must have all array members as a single word i.e. "${name[*]}"
+# because it should succeed when there is any non-empty array member, not necessarily the first one:
+test "${CLONE_USERS[*]}" && Log "Cloning users: ${CLONE_USERS[@]}"
+for user in "${CLONE_USERS[@]}" ; do
+    # Skip if the user exists already in the ReaR recovery system:
+    grep "^$user:" $ROOTFS_DIR/etc/passwd 1>&2 && continue
+    # Skip if the user does not exist in the current system:
+    if ! passwd_entry="$( getent passwd $user )" ; then
+        Debug "Cannot clone $user because it does not exist"
+        continue
+    fi
+    # Prepare to also add the user's group to the CLONE_GROUPS array:
+    # passwd_entry="user:password:UID:GID:description:HOMEdirectory:shell
+    groupID="$( echo "$passwd_entry" | cut -d ':' -f '4' )"
```

Comment:
You could use `cut ... <<<"$passwd_entry"` to manage without the pipe

Reply:
Yes, thanks!
I am not yet used to use '<<<'.